python/ml4ir/applications/classification/tests/test_classification_serving.py

```diff
@@ -0,0 +1,119 @@
+        # SequenceExample inputs are supported and we face https://github.com/tensorflow/tensorflow/issues/31686. #
+        model_file = os.path.join(self.args.models_dir, "final")                                                  #
+        # Save model with default signature                                                                       #
+        classification_model.model.save(filepath=os.path.join(model_file, "default"))                             #
+        ###########################################################################################################
+
+        default_model = kmodels.load_model(
+            os.path.join(self.output_dir, "final", "default"), compile=False
+        )
+        assert ServingSignatureKey.DEFAULT in default_model.signatures
+        default_signature = default_model.signatures[ServingSignatureKey.DEFAULT]
+
+        # Fetch a single batch for testing
+        parsed_sequence_examples = next(iter(relevance_dataset.test))[0]
+        parsed_dataset_batch = relevance_dataset.test.take(1)
+
+        # Use the loaded serving signatures for inference
+        model_predictions = classification_model.predict(parsed_dataset_batch)[self.args.output_name].values
+        default_signature_predictions = default_signature(**parsed_sequence_examples)[
+            self.args.output_name
+        ]
+
+        # Compare the scores from the different versions of the model
+        assert np.isclose(model_predictions[0], default_signature_predictions[0], rtol=0.01, ).all()
+
+    def get_tfrecord_signature(self):
+        self.set_seeds()
+        classification_pipeline: ClassificationPipeline = ClassificationPipeline(args=self.get_overridden_args())
+
+        classification_model: RelevanceModel = classification_pipeline.get_relevance_model()
+
+        classification_model.save(
+            models_dir=self.args.models_dir,
+            preprocessing_keys_to_fns={},
+            postprocessing_fn=None,
+            required_fields_only=not self.args.use_all_fields_at_inference,
+            pad_sequence=self.args.pad_sequence_at_inference,
+        )
+
+        # Load SavedModel and get the right serving signature
+        tfrecord_model = kmodels.load_model(
+            os.path.join(self.output_dir, "final", "tfrecord"), compile=False
+        )
+        assert ServingSignatureKey.TFRECORD in tfrecord_model.signatures
+
+        return tfrecord_model.signatures[ServingSignatureKey.TFRECORD]
+
+    def get_feature_config(self):
+        feature_config: FeatureConfig = FeatureConfig.get_instance(
+            tfrecord_type=self.args.tfrecord_type,
+            feature_config_dict=self.feature_config,
+            logger=self.logger,
+        )
+
+        return feature_config
+
+####################################################################################################
+# Test below disabled as we currently face: https://github.com/tensorflow/tensorflow/issues/31686. #
```

Comment:
@Ullimague Can you please paste your stacktrace? The issue you linked has nothing to do with this scenario. Where are you using categorical_column_with_vocabulary_list?